ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/ImmDB.hs

```diff
@@ -230,6 +232,52 @@ mkImmDB immDB decHeader decBlock encBlock epochInfo isEBB addHdrEnv err = ImmDB
   Getting and parsing blocks
 -------------------------------------------------------------------------------}
 
+-- | Return 'True' when the given point is in the ImmutableDB.
+--
+-- This is much more efficient than 'getBlockOrHeaderWithPoint' as no block or
+-- header has to be read from disk.
+--
+-- If the point corresponds to some slot in the future, 'False' is returned.
```

Comment:
It would be good to document why this change in behaviour compared to other functions is necessary (wrt future slots).

Because otherwise we could just use `GetWhat` here instantiated to hash or something like that, rather than introducing yet another function? (It's possible that this will need the `BlockComponent` changes first.)

Reply:
Hmm, it returns `False` instead of throwing a `ReadFuture*Error` because actually checking whether the block/EBB is not in the future is non-trivial and involves converting to/from slots/epochs. The former implementation of `getBlock(OrHeader)WithPoint` used to do it, but the current implementation no longer does it.

So I'm afraid that we can't simply use `GetHash` here, as it won't check the bounds. Note that we can't know upfront whether a `Point` is in the future, we have to access the ImmutableDB for that, in particular to know whether the `Point` corresponds to a block or slot. Alternatively, we could just catch the `ReadFuture*Error`. Would you be okay with that?